docs/rfc/0001-control-api.md

```diff
@@ -0,0 +1,991 @@
+        on_leave: "grpc://127.0.0.1:9091"
+        pipeline:
+          # Media element which is able to receive media data from client via WebRTC.
+          publish:
+            kind: WebRtcPublishEndpoint
+            spec:
+              # Actually, it receives not media data, but ICE candidates only.
+              p2p: Always
+              # Fires when "caller" client starts publishing media data.
+              on_start: "http://127.0.0.1:8080/publish/started"
+              # Fires when "caller" client stops publishing media data.
+              on_stop: "http://127.0.0.1:8080/publish/stopped"
+          # Media element which is able to play media data for client via WebRTC.
+          play:
+            kind: WebRtcPlayEndpoint
+            spec:
+              # The source to get media data from.
+              # It will take responder's ICE candidates and propose them to client side,
+              # so the WebRTC negotiation can happen.
+              src: "local://video-call-2/responder/publish"
+              # Fires when "caller" client starts playing media data of "responder" client.
+              on_start: "http://127.0.0.1:8080/play/started"
+              # Fires when "caller" client stops playing media data of "responder" client.
+              on_stop: "http://127.0.0.1:8080/play/stopped"
+    responder:
+      kind: Member
+      spec:
+        on_join: "grpc://127.0.0.1:9091"
+        on_leave: "grpc://127.0.0.1:9091"
+        pipeline:
+          publish:
+            kind: WebRtcPublishEndpoint
+            spec:
+              p2p: Always
+              on_start: "http://127.0.0.1:8080/publish/started"
+              on_stop: "http://127.0.0.1:8080/publish/stopped"
+          play:
+            kind: WebRtcPlayEndpoint
+            spec:
+              src: "local://video-call-2/caller/publish"
+              on_start: "http://127.0.0.1:8080/play/started"
+              on_stop: "http://127.0.0.1:8080/play/stopped"
+```
+</details>
+
+After creation of desired media data pipeline via Control API, it can be used from the client side.
+
+Depending on concrete `Element`'s kind, it may implement auto-removing rules, which are going to help avoiding manual `Element`s removing in some situations (remove the whole `Room` automatically when video call ends, for example). These rules may vary, depending on `Element`'s behavior, but commonly have the following semantics (`Element` is auto-removed when):
+- all real clients (`Member`s) have left the media server;
+- all media data sources have ended.
+
+
+
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+For better integration capabilities the Control API should be implemented in two flavors (both can be enabled/disabled/configured via media server configuration):
+- *[gRPC] interface*, which easy-to-go due to code generation;
+- *HTTP rest interface* for languages and tools which are not supported by [gRPC].
```

Comment:
@tyranron ,

А, может, на [protoc-gen-swagger](https://github.com/grpc-ecosystem/grpc-gateway/tree/master/protoc-gen-swagger) посмотрим?

Reply:
@alexlapa `proto/grpc` is used because it's no way to describe gRPC interface normally without the spec. With REST HTTP it's easy as we see.

I may mention Swagger in a reference-level guide as a possible tool, but not very motivated to write any Swagger specs in the RFC.